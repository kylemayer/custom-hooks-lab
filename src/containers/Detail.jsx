import React from 'react';
import CharacterDetail from '../components/displays/CharacterDetail';
import { Link } from 'react-router-dom';
import { useCharacterById } from '../hooks/useCharacterById';

const Detail = () => {
  const { loading, character } = useCharacterById();

  if (loading) return <h4>Loading...</h4>;

  return (
    <>
      <Link to="/">
        <h3>Go Back</h3>
      </Link>
      <CharacterDetail name={character.name} image={character.image} />
    </>
  );
};

export default Detail;
