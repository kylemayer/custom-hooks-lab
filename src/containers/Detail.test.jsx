/* eslint-disable max-len */
import React from 'react';
import { render, screen } from '@testing-library/react';
import Detail from './Detail';
import { MemoryRouter } from 'react-router-dom';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import character from '../json/character.json';

const server = setupServer(
  rest.get(
    'https://hey-arnold-api.herokuapp.com/api/v1/characters/:id',
    (req, res, ctx) => {
      return res(ctx.json(character));
    }
  )
);

describe('HeyArnold detail container', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('displays Nancy character', async () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/5da237699734fdcb7bef8f5c']} keyLength={25}>
        <Detail />
      </MemoryRouter>
    );

    screen.getByText('Loading...');

    await screen.findByText('Nancy', { exact: false });
    expect(container).toMatchSnapshot();
  });
});
