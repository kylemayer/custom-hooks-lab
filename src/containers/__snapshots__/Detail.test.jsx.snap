// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeyArnold detail container displays Nancy character 1`] = `
<div>
  <a
    href="/"
  >
    <h3>
      Go Back
    </h3>
  </a>
  <div
    role="character-detail"
  >
    <figure>
      <img
        alt="Nancy"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
      />
      <figcaption>
        Nancy
      </figcaption>
    </figure>
  </div>
</div>
`;
